models/node_helpers.go

```diff
@@ -80,9 +80,19 @@ func checkUniqueNodeInstanceRegion(q *reform.Querier, instance, region string) e
 	}
 }
 
-// FindAllNodes returns all Nodes.
-func FindAllNodes(q *reform.Querier) ([]*Node, error) {
-	structs, err := q.SelectAllFrom(NodeTable, "ORDER BY node_id")
+// FindNodes returns Nodes by filters.
+func FindNodes(q *reform.Querier, filters NodeFilters) ([]*Node, error) {
+	var conditions []string
+	var args []interface{}
+	if filters.NodeType != nil {
+		conditions = append(conditions, "node_type = $1")
+		args = append(args, filters.NodeType)
+	}
+	var whereClause string
+	if len(conditions) != 0 {
+		whereClause = fmt.Sprintf("WHERE %s", strings.Join(conditions, " AND "))
+	}
+	structs, err := q.SelectAllFrom(NodeTable, fmt.Sprintf("%s ORDER BY node_id", whereClause), args...)
```

Comment:
I see it looks like follow a general pattern, but here it a bit overcomplication.
Maybe it could be simplified somehow like this? 
```suggestion
	var args []interface{}
	tail := "ORDER BY node_id"
	if filters.NodeType != nil {
	        tail = "WHERE node_type = $1 " + tail
	        args = []interface{}{filters.NodeType}
	}
	structs, err := q.SelectAllFrom(NodeTable, tail, args...)
```